### PR TITLE
run http2-frame-test-case test cases and fix padding

### LINKED ISSUE #68: Test against http2jp/http2-frame-test-case
Similarly to python-hyper/hpack#76, we should test against the [http2jp/http2-frame-test-case](https://github.com/http2jp/http2-frame-test-case) test data to verify implementation.

(Unlike the hpack one, there is nothing to contribute back)

### PR DESCRIPTION
**Pull request by Kriechi**
fixes #68 

This PR runs the https://github.com/http2jp/http2-frame-test-case test cases as git-submodule with a small pytest wrapper to load them dynamically.

This PR fixes and actual serious bug with regard to padding in PUSH_PROMISE frames.

This PR contains various commits to fix individual bugs or test failures for frame parsing. Most of them are about increased input validation and more strict/correct interpretation of the RFC.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,4 +34,4 @@ jobs:
         tox --parallel auto --notest
     - name: Test with tox
       run: |
-        tox --parallel 0
+        tox --parallel 0 -v
```
Comment (pgjones) [inline .github/workflows/ci.yml:37]: @Kriechi Do you mean to keep this, or was it for your development?

File: HISTORY.rst [hunk 2]
```diff
@@ -4,6 +4,23 @@ Release History
...
+- Fixed padding parsing for ``PushPromiseFrame``.
+- Fixed unchecked frame length for ``PriorityFrame``. It now correctly raises ``InvalidFrameError``.
+- Fixed promised stream id parsing for ``PushPromiseFrame``.
+- Fixed unchecked frame length for ``WindowUpdateFrame``. It now correctly raises ``InvalidFrameError``.
+- Fixed window increment value range validation. It must be 1 <= increment <= 2^31-1.
+- Fixed parsing of ``SettingsFrame`` with mutual exclusion of ACK flag and payload.
+
+**Other Changes**
+
+- Drop support for Python 2.7, 3.4, 3.5, mypy and support 3.8.
```
Comment (pgjones) [inline HISTORY.rst:22]: Do you mean `mypy` here? (Or pypy, or support mypy linting?)

File: hyperframe/frame.py [hunk 3]
```diff
@@ -316,8 +323,14 @@ def serialize_body(self):
...
     def parse_body(self, data):
+        if len(data) > 5:
+            raise InvalidFrameError(
+                "PRIORITY must have 5 byte body: actual length %s." %
+                len(data)
+            )
+
         self.parse_priority_data(data)
-        self.body_len = len(data)
+        self.body_len = 5
```
Comment (pgjones) [inline hyperframe/frame.py:333]: Could you put the 5 as a CONSTANT at the top of the file? Seems to match the style used previously. Also allows the commenting convention.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Kriechi) [reply .github/workflows/ci.yml:37]: dev work - I remove it.

File: HISTORY.rst (hunk 2 above)
Comment (Kriechi) [reply HISTORY.rst:22]: Thats a typo - support for pypy was dropped (because it's py27).

File: hyperframe/frame.py (hunk 3 above)
Comment (Kriechi) [reply hyperframe/frame.py:333]: Which style does it match?
All fixed-length frames have a "self.body_len = XYZ" statement:
RstStreamFrame, PingFrame, WindowUpdateFrame

Comment (pgjones) [reply hyperframe/frame.py:333]: Oh yea, fair enough.

**Pull request merged**